src/webgl/ShaderGenerator.js

```diff
@@ -1638,3 +1638,509 @@ export default shadergenerator;
+ * - <a href="#/p5/baseColorShader">baseColorShader()</a>
+ * - <a href="#/p5/baseMaterialShader">baseMaterialShader()</a>
+ * - <a href="#/p5/baseNormalShader">baseNormalShader()</a>
+ * - <a href="#/p5/baseStrokeShader">baseStrokeShader()</a>
+ *
+ * @param {Function} callback
+ *        A callback function which receives the vertex object and should return it after making any changes.
+ *
+ * @example
+ * <div modernizr='webgl'>
+ * <code>
+ * let myShader;
+ * function setup() {
+ *   createCanvas(200, 200, WEBGL);
+ *   myShader = baseColorShader().modify(() => {
+ *     let t = uniformFloat(() => millis());
+ *     getObjectInputs(inputs => {
+ *       // Create a sine wave along the x axis in object space
+ *       inputs.position.y += 3 * sin(t * 0.001 + inputs.position.x * 0.05);
+ *       return inputs;
+ *     });
+ *   });
+ * }
+ * function draw() {
+ *   background(220);
+ *   shader(myShader);
+ *   noStroke();
+ *   fill('orange');
+ *   sphere(100);
+ * }
+ * </code>
+ * </div>
+ */
+
+/**
+ * @method getCameraInputs
+ * @description
+ * Registers a callback to adjust vertex properties after the model has been transformed by the camera, but before projection, in the vertex shader. This hook can be used inside <a href="#/p5/baseColorShader">baseColorShader()</a>.modify() and similar shader <a href="#/p5.Shader/modify">modify()</a> calls to create effects that depend on the camera's view. The callback receives an object with the following properties:
+ * 
+ * - `position`: a three-component vector representing the position after camera transformation.
+ * - `normal`: a three-component vector representing the normal after camera transformation.
+ * - `texCoord`: a two-component vector representing the texture coordinates.
+ * - `color`: a four-component vector representing the color of the vertex (red, green, blue, alpha).
+ *
+ * Return the modified object to update the vertex.
+ *
+ * This hook is available in:
+ * - <a href="#/p5/baseColorShader">baseColorShader()</a>
+ * - <a href="#/p5/baseMaterialShader">baseMaterialShader()</a>
+ * - <a href="#/p5/baseNormalShader">baseNormalShader()</a>
+ * - <a href="#/p5/baseStrokeShader">baseStrokeShader()</a>
+ *
+ * @param {Function} callback
+ *        A callback function which receives the vertex object and should return it after making any changes.
+ *
+ * @example
+ * <div modernizr='webgl'>
+ * <code>
+ * let myShader;
+ * function setup() {
+ *   createCanvas(200, 200, WEBGL);
+ *   myShader = baseColorShader().modify(() => {
+ *     getCameraInputs(inputs => {
+ *       // Move vertices in camera space based on their x position
+ *       let t = uniformFloat(() => millis());
+ *       inputs.position.y += 30 * sin(inputs.position.x * 0.05 + t * 0.001);
+ *       // Tint all vertices blue
+ *       inputs.color.b = 1;
+ *       return inputs;
+ *     });
+ *   });
+ * }
+ * function draw() {
+ *   background(200);
+ *   shader(myShader);
+ *   noStroke();
+ *   fill('blue');
```

Comment:
Currently the `inputs.color.b = 1` has no effect if you comment it out because the input color's blue channel is already maxed out from the `fill('blue')` here. Can we use `fill('red')` so that we get a visible color change?